src/platform_impl/web/web_sys/pointer.rs

```diff
@@ -31,133 +32,152 @@ impl PointerHandler {
                     ),
                     "mouse" => mouse_handler(
                         event.pointer_id(),
+                        event::mouse_position(&event).to_physical(super::scale_factor()),
                         event::mouse_button(&event).expect("no mouse button released"),
-                        event::mouse_modifiers(&event),
                     ),
                     _ => (),
                 }
             },
         ));
     }
 
-    pub fn on_mouse_press<M, T>(
+    pub fn on_mouse_press<MOD, M, T>(
         &mut self,
-        canvas_common: &super::Common,
+        canvas_common: &Common,
+        mut modifier_handler: MOD,
         mut mouse_handler: M,
         mut touch_handler: T,
+        prevent_default: bool,
     ) where
-        M: 'static + FnMut(i32, PhysicalPosition<f64>, MouseButton, ModifiersState),
+        MOD: 'static + FnMut(ModifiersState),
+        M: 'static + FnMut(i32, PhysicalPosition<f64>, MouseButton),
         T: 'static + FnMut(i32, PhysicalPosition<f64>, Force),
     {
         let canvas = canvas_common.raw.clone();
         self.on_pointer_press = Some(canvas_common.add_user_event(
             "pointerdown",
             move |event: PointerEvent| {
+                if prevent_default {
+                    // prevent text selection
+                    event.prevent_default();
+                    // but still focus element
+                    let _ = canvas.focus();
+                }
+
+                modifier_handler(event::mouse_modifiers(&event));
+
                 match event.pointer_type().as_str() {
                     "touch" => {
                         touch_handler(
                             event.pointer_id(),
                             event::touch_position(&event, &canvas)
-                                .to_physical(super::super::scale_factor()),
+                                .to_physical(super::scale_factor()),
                             Force::Normalized(event.pressure() as f64),
                         );
                     }
                     "mouse" => {
                         mouse_handler(
                             event.pointer_id(),
-                            event::mouse_position(&event).to_physical(super::super::scale_factor()),
+                            event::mouse_position(&event).to_physical(super::scale_factor()),
                             event::mouse_button(&event).expect("no mouse button pressed"),
-                            event::mouse_modifiers(&event),
                         );
-
-                        // Error is swallowed here since the error would occur every time the mouse is
-                        // clicked when the cursor is grabbed, and there is probably not a situation where
-                        // this could fail, that we care if it fails.
-                        let _e = canvas.set_pointer_capture(event.pointer_id());
```

Comment:
Hey @ryo33!
Do you still remember why you added this in https://github.com/rust-windowing/winit/pull/2188?

Reply:
https://github.com/rust-windowing/winit/commit/c5c99d2357880792a1f552a83eeb67f074103688
It seems added two years ago in this commit. The commit message says the same as the comment:

> * Remove "expect" from fallible code
>
>    code would crash if handling pointer capture outside of winit on
>    every mouse click
>
>    we swallow the error since we could not think of a case where this
>    would fail in a way that would want to handle that it fails

Reply:
I was talking about the call to `setPointerCapture()`, not why the error is ignored.
In any case, my bad, I just didn't go back far enough: https://github.com/rust-windowing/winit/pull/1672; which addressed https://github.com/rust-windowing/winit/issues/1660.

Now that makes perfect sense, I should definitely not remove that. Will figure out how exactly to sync this with other backends.

Thanks!